Xamarin.Forms.Platform.iOS/Renderers/ListViewRenderer.cs

```diff
@@ -1175,11 +1175,14 @@ public override void RowSelected(UITableView tableView, NSIndexPath indexPath)
 
 			public override void WillDisplay(UITableView tableView, UITableViewCell cell, NSIndexPath indexPath)
 			{
-				if (!_estimatedRowHeight)
+				Device.BeginInvokeOnMainThread(() =>
```

Comment:
Why does queueing this up on the main thread fix the issue?

Reply:
To be perfectly honest: I don't know exactly. The problem goes away if you take out [`WillDisplay`](https://github.com/xamarin/Xamarin.Forms/pull/6576/files#diff-198607f4ce97efcc515a271473524b08R1176) as a whole. So, when debugging through that, I noticed that the `WillDisplay` was called several times and each time `_estimatedRowHeight` was `false`.

Here I began to think that it was some kind of threading/race condition thing. I tried setting the `_estimatedRowHeight` to true earlier on and some other variations on that, but the only thing that seemed to help in the end was making sure it synced with the main thread. If you have any other suggestions, I'm happy to try them!

Reply:
@samhouts seems to think this is a regression: https://github.com/xamarin/Xamarin.Forms/issues/6472#issuecomment-501895542

Do we know which commit regressed it? That might give us a clue as to the real cause.

Reply:
I’ll dig around a little, from what I remember that is because this is something Rui implemented somewhere in between to get the right rowheight. So this code wasn’t in 3.5 but added towards 3.6.